File: python/pallindrome_str.py
#Optimized sol
def is_palindrome(string):
    palindrome = True
    i = 0
    j = len(string)-1
    if (len(string)%2 == 0) :
        print("even")
        while (i<j) :
            print(i,"  ",j)
            if (string[i]==string[j]) :               
                i+=1
                j-=1
            else:
                palindrome = False
                break
    else:
        print("odd")
        while (i<=j) :
            print(i,"  ",j)
            if (string[i]==string[j]):             
                 i+=1
                 j-=1
            else:
                 palindrome = False
                 break
    return palindrome
print(is_palindrome('a1a'))

File: javascript/pallindrome_str.js
//Optimal
// function is_palindrome(string){
//     const hashTable = {};
//     for(let i=0;i<string.length;i++){
//         if(string[i] in hashTable){
//             hashTable[string[i]]++;
//         } else{
//             hashTable[string[i]]=1;
//         }
//     }
//     for(let i=0;i<string.length;i++){
//         if(hashTable[string[i]]===1){
//             return i;
//         }
//     }
//     return null;
// }

//brute force
function is_palindrome(string){
    let palindrome = true;
    let i = 0
    let j = string.length-1
    if (string.length%2 === 0) {
        console.log("even");
        while (i<j) {
            console.log(i,"  ",j);
           if (string[i]===string[j]) {                
                i++
                j--
           }else{
                palindrome = false
                break
           } 
        }
    } else {
        console.log("odd");
        while (i<=j) {
            console.log(i,"  ",j);
            if (string[i]===string[j]) {                
                 i++
                 j--
            }else{
                 palindrome = false
                 break
            } 
         }
    }
    return palindrome
}
a='a12421a';
console.log(is_palindrome(a));